src/htm/algorithms/Connections.cpp

```diff
@@ -206,18 +206,11 @@ void Connections::destroySegment(const Segment segment) {
 
   CellData &cellData = cells_[segmentData.cell];
 
-  const auto segmentOnCell =
-      std::lower_bound(cellData.segments.cbegin(), cellData.segments.cend(),
-                       segment, 
-		       [&](const Segment a, const Segment b) {
-			 return dataForSegment(a).id < dataForSegment(b).id;
-                       });
-
-  NTA_ASSERT(segmentOnCell != cellData.segments.end());
+  const auto segmentOnCell = std::find(cellData.segments.cbegin(), cellData.segments.cend(), segment);
```

Comment:
the error was dumb, comparing const iterator above with `.segments.end()` here. 
Replaced with a simpler `find` that does not require extra sorting.